test/unit/org/apache/cassandra/cql3/validation/operations/BatchTest.java

```diff
@@ -55,24 +55,53 @@ public void testBatchAndList() throws Throwable
         createTable("CREATE TABLE %s (k int PRIMARY KEY, l list<int>)");
 
         execute("BEGIN BATCH " +
-                "UPDATE %1$s SET l = l +[ 1 ] WHERE k = 0; " +
-                "UPDATE %1$s SET l = l + [ 2 ] WHERE k = 0; " +
-                "UPDATE %1$s SET l = l + [ 3 ] WHERE k = 0; " +
+                "UPDATE %1$s SET l = l + [1] WHERE k = 0; " +
+                "UPDATE %1$s SET l = l + [2] WHERE k = 0; " +
+                "UPDATE %1$s SET l = l + [3] WHERE k = 0; " +
                 "APPLY BATCH");
 
         assertRows(execute("SELECT l FROM %s WHERE k = 0"),
                    row(list(1, 2, 3)));
 
         execute("BEGIN BATCH " +
-                "UPDATE %1$s SET l =[ 1 ] + l WHERE k = 1; " +
-                "UPDATE %1$s SET l = [ 2 ] + l WHERE k = 1; " +
-                "UPDATE %1$s SET l = [ 3 ] + l WHERE k = 1; " +
+                "UPDATE %1$s SET l = [1] + l WHERE k = 1; " +
+                "UPDATE %1$s SET l = [2] + l WHERE k = 1; " +
+                "UPDATE %1$s SET l = [3] + l WHERE k = 1; " +
                 "APPLY BATCH ");
 
         assertRows(execute("SELECT l FROM %s WHERE k = 1"),
                    row(list(3, 2, 1)));
     }
 
+    @Test
+    public void testBatchAndMap() throws Throwable
+    {
+        createTable("CREATE TABLE %s (k int PRIMARY KEY, m map<int, int>)");
+
+        execute("BEGIN BATCH " +
+                "UPDATE %1$s SET m[1] = 1 WHERE k = 0; " +
+                "UPDATE %1$s SET m[2] = 2 WHERE k = 0; " +
```

Comment:
Nit: we could use different values for keys and values, to detect if they are somehow confused with each other, for example:
```java
execute("BEGIN BATCH " +
        "UPDATE %1$s SET m[1] = 10 WHERE k = 0; " +
        "UPDATE %1$s SET m[2] = 20 WHERE k = 0; " +
        "APPLY BATCH");

assertRows(execute("SELECT m FROM %s WHERE k = 0"),
           row(map(1, 10, 2, 20)));
```